src/main/kotlin/su/arlet/business1/security/jwt/AuthTokenFilter.kt

```diff
@@ -0,0 +1,69 @@
+package su.arlet.business1.security.jwt
+
+import jakarta.servlet.FilterChain
+import jakarta.servlet.ServletException
+import jakarta.servlet.http.HttpServletRequest
+import jakarta.servlet.http.HttpServletResponse
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
+import org.springframework.security.core.context.SecurityContextHolder
+import org.springframework.security.core.userdetails.UserDetails
+import org.springframework.security.core.userdetails.UsernameNotFoundException
+import org.springframework.security.web.authentication.WebAuthenticationDetailsSource
+import org.springframework.util.StringUtils
+import org.springframework.web.filter.OncePerRequestFilter
+import su.arlet.business1.exceptions.UserNotFoundException
+import su.arlet.business1.security.sevices.AuthUsersDetails
+import su.arlet.business1.security.sevices.AuthUsersDetailsService
+import java.io.IOException
+
+class AuthTokenFilter: OncePerRequestFilter() {
+    @Autowired
+    private lateinit var jwtUtils: JwtUtils
+    @Autowired
+    private lateinit var userDetailsService: AuthUsersDetailsService
+
+    @Throws(IOException::class, ServletException::class)
+    override fun doFilterInternal(
+        request: HttpServletRequest,
+        response: HttpServletResponse,
+        filterChain: FilterChain
+    ) {
+        try {
+            val userDetails = getUserDetails(request)
+            if (userDetails != null) {
+                val authentication = UsernamePasswordAuthenticationToken(
+                    userDetails, null, userDetails.authorities
+                )
+                authentication.details = WebAuthenticationDetailsSource().buildDetails(request)
+                SecurityContextHolder.getContext().authentication = authentication
+            }
+        } catch (e: Exception) {
```

Comment:
Need to handle user not found exception cause it's a filter and controller advice cannot help us for it